applications/crossbar/src/api_util.erl

```diff
@@ -720,6 +736,30 @@ is_permitted_nouns(Req, Context0, _Nouns) ->
             ?MODULE:halt(Req, Context1)
     end.
 
+-spec is_permitted_nouns(cowboy_req:req(), cb_context:context(), http_method(), list()) ->
+                                {'true', cowboy_req:req(), cb_context:context()} |
+                                halt_return().
+is_permitted_nouns(Req, Context, _ReqVerb, [{<<"404">>, []}]) ->
+    ?MODULE:halt(Req, cb_context:add_system_error('not_found', Context));
+is_permitted_nouns(Req, Context, _ReqVerb, []) ->
+    lager:debug("no one authz'd the request"),
+    ?MODULE:halt(Req, cb_context:add_system_error('forbidden', Context));
+is_permitted_nouns(Req, Context0, _ReqVerb, [{Mod, Params} | ReqNouns]) ->
+    Event = api_util:create_event_name(Context0, <<"authorize.", Mod/binary>>),
+    Payload = [Context0 | Params],
+    case crossbar_bindings:succeeded(crossbar_bindings:map(Event, Payload)) of
+        [] ->
+            lager:debug("failed to authorize : ~p", [Mod]),
+            is_permitted_nouns(Req, Context0, _ReqVerb, ReqNouns);
```

Comment:
I dont think we should loop the ReqNouns.  Contrived example but this would cause the auth for:

v1/accounts/123/search
as well as
v1/search

which may not be expected.  I think using the final token, which is the end of the URL chain, is expected (as we do for execute).